lib/internal/util.js

```diff
@@ -292,10 +292,16 @@ function getIdentificationOf(obj) {
 const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
 const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
 
-function promisify(original) {
+function promisify(original, options = {}) {
   if (typeof original !== 'function')
     throw new ERR_INVALID_ARG_TYPE('original', 'Function');
 
+  if (Object.prototype.toString.call(options) !== '[object Object]') {
```

Comment:
`if (typeof options !== 'object' || options == null)`

Reply:
but `typeof []` is `options` too

Reply:
@jasnell infact, making that change will cause a test I wrote to fail.

Reply:
> but `typeof [1, 2]` is `object` too

I'm not sure that's a problem.

```js
const options = [1, 2];
options.context = someContext;
util.promisify(someFunction, options);
```